js/test/wallet/createTransaction.test.ts

```diff
@@ -0,0 +1,436 @@
+                }).rejects.toThrow('payer frozen');
+            });
+
+            it('throws an error on unitialized recipient', async () => {
+                expect.assertions(1);
+
+                mockGetAccount.mockImplementation((_, ATA) => {
+                    switch (ATA.toString()) {
+                        case payerATA.toString():
+                            return { isInitialized: true, isFrozen: false };
+
+                        default:
+                            return { isInitialized: false, isFrozen: false };
+                    }
+                });
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        recipientPublicKey,
+                        new BigNumber(100),
+                        {
+                            token: new PublicKey('8FRFC6MoGGkMFQwngccyu69VnYbzykGeez7ignHVAFSN'),
+                        }
+                    );
+                }).rejects.toThrow('recipient not initialized');
+            });
+
+            it('throws an error on frozen recipient', async () => {
+                expect.assertions(1);
+
+                mockGetAccount.mockImplementation((_, ATA) => {
+                    switch (ATA.toString()) {
+                        case payerATA.toString():
+                            return { isInitialized: true, isFrozen: false };
+
+                        default:
+                            return { isInitialized: true, isFrozen: true };
+                    }
+                });
+
+                await expect(async () => {
+                    return await createTransaction(
+                        connection,
+                        wallet.publicKey,
+                        recipientPublicKey,
+                        new BigNumber(100),
+                        {
+                            token: new PublicKey('8FRFC6MoGGkMFQwngccyu69VnYbzykGeez7ignHVAFSN'),
+                        }
+                    );
+                }).rejects.toThrow('recipient frozen');
+            });
+
+            it('throws an error on insufficient funds', async () => {
+                expect.assertions(1);
+
+                await expect(async () => {
+                    return await createTransaction(connection, wallet.publicKey, recipientPublicKey, new BigNumber(1), {
+                        token: new PublicKey('mvines9iiHiQTysrwkJjGf2gb9Ex9jXJX8ns3qwf2kN'),
+                    });
+                }).rejects.toThrow('insufficient funds');
+            });
+        });
+    });
+});
+
+/**
+ * Test helper to create a SOL transaction.
+ */
+function createSOLTransaction(
+    payer: PublicKey,
+    recipient: PublicKey,
+    amount: BigNumber,
+    {
+        token,
+        references,
+        memo,
+    }: {
+        token?: PublicKey;
+        references?: PublicKey[];
+        memo?: string;
+    }
+): Transaction {
```

Comment:
This helper function merely duplicates the entire logic between the test code and the source code. This isn't really testing what we want.

It's better to just serialize the expected transaction (as JSON or bytes or base58 or something) to deterministically test for regression.